### PR TITLE
Set up a "Hello world" project as one app with Webpack

### PR DESCRIPTION
**Pull request by akos29**
Set up a "Hello world" project as one app with Webpack
--

This pull request adds a "Hello World" project named hello-rails-react that integrates React for the frontend and uses Postgres for the database. The project includes the following milestones and steps:

- Created a new Rails app named hello-rails-react that incorporates React for the frontend and Postgres for the database.
- Set up a database table to store messages and populated it with 5 different greeting messages.
- Added an API endpoint that selects a random greeting message from the database. 
- Created a static view as the root of the app to serve as the entry point. 
- Developed an App component using react-router and rendered it within the static view.
- Created a Greeting component responsible for displaying a random greeting message, and set it up as a route within the App component.
- Implemented a Redux store, an action, and a reducer to communicate with the API endpoint and retrieve a random greeting message. 
- Displayed the fetched random greeting message within the Greeting component.

### CODE REVIEW

**Review by omar-muhamad — COMMENTED** [review]
# Invalid ❌

Hi @akos29,

#### While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

#### Your Code Review Request is marked as invalid for this reason:

- [ ] You didn't document your work professionally:
   - You used the Microverse README template, but you forgot to edit it. Please edit the template with your personal and project data.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **Does not count** into the code reviews limit._

File: babel.config.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  const resultConfig = defaultConfigFunc(api);
+
+  const changesOnDefault = {
+    plugins: [process.env.WEBPACK_SERVE && 'react-refresh/babel'].filter(Boolean),
+  };
+
+  resultConfig.plugins = [...resultConfig.plugins, ...changesOnDefault.plugins];
+
+  return resultConfig;
+};
```
Comment (FrederickMih) [inline babel.config.js]: - [ ] Based on the requirements, config files should be set up as shown in[ this tutorial](https://betterprogramming.pub/react-with-rails-2022-bd28e1fcd355#ad0a). I got the error below when trying to run your app locally

Screenshot from 2023-09-06 17-14-24

File: app/javascript/components/App.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import store from './store';
+import Greeting from './components/Greeting';
+
+const Home = () => <div>Home Page</div>;
+
+function App() {
+  return (
+    <Provider store={store}>
+      <Router>
+        <Switch>
```
Comment (FrederickMih) [inline app/javascript/components/App.js]: - [ ] Your app could not be run successfully as a result of this error

Screenshot from 2023-09-06 17-18-15

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: :raising_hand: @akos29, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!





## Highlights :green_circle: 

- Required app dependencies were set up :+1: 


## Required Changes ♻️
**N/A**
_Check the comments under the review._

## Optional suggestions
**N/A**
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AmaduKamara — APPROVED** [review]
# Status: Approved 🟢 🟢 🟢 

Hi @akos29,

Your project is complete!  There is nothing else to say other than... It's time to merge it: ship it:
Congratulations! 🎉 

## Highlights:

✅ You set up Gitflow correctly 👍🏽
✅ The Linter checks passed successfully 👍🏽
✅ The rails server runs fine and renders random greetings 👍🏽
✅ Set up Redux correctly 👍🏽
✅ The React app renders random greetings as expected 👍🏽
✅ Good descriptive PR and commit messages 👍🏽

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**